external/types/pdf.d.ts

```diff
@@ -0,0 +1,112 @@
+
+declare module 'pdfjs-dist' {
+  //
+  // src/pdf.js
+  //
+  export import build = _pdfjs.build;
+  export import version = _pdfjs.version;
+  export import getDocument = _pdfjs.getDocument;
+  export import LoopbackPort = _pdfjs.LoopbackPort;
+  export import PDFDataRangeTransport = _pdfjs.PDFDataRangeTransport;
+  export import PDFWorker = _pdfjs.PDFWorker;
+  export import renderTextLayer = _pdfjs.renderTextLayer;
+  export import AnnotationLayer = _pdfjs.AnnotationLayer;
+  export import createPromiseCapability = _pdfjs.createPromiseCapability;
+  export import PasswordResponses = _pdfjs.PasswordResponses;
+  export import InvalidPDFException = _pdfjs.InvalidPDFException;
+  export import MissingPDFException = _pdfjs.MissingPDFException;
+  export import SVGGraphics = _pdfjs.SVGGraphics;
+  export import NativeImageDecoding = _pdfjs.NativeImageDecoding;
+  export import UnexpectedResponseException = _pdfjs.UnexpectedResponseException;
+  export import OPS = _pdfjs.OPS;
+  export import VerbosityLevel = _pdfjs.VerbosityLevel;
+  export import UNSUPPORTED_FEATURES = _pdfjs.UNSUPPORTED_FEATURES;
+  export import createValidAbsoluteUrl = _pdfjs.createValidAbsoluteUrl;
+  export import createObjectURL = _pdfjs.createObjectURL;
+  export import removeNullCharacters = _pdfjs.removeNullCharacters;
+  export import shadow = _pdfjs.shadow;
+  export import createBlob = _pdfjs.createBlob;
+  export import Util = _pdfjs.Util;
+  export import RenderingCancelledException = _pdfjs.RenderingCancelledException;
+  export import getFilenameFromUrl = _pdfjs.getFilenameFromUrl;
+  export import LinkTarget = _pdfjs.LinkTarget;
+  export import addLinkAttributes = _pdfjs.addLinkAttributes;
+  export import GlobalWorkerOptions = _pdfjs.GlobalWorkerOptions;
+  export import apiCompatibilityParams = _pdfjs.apiCompatibilityParams;
+
+  //
+  // Convenience types that are used often.
+  //
+  export import PDFDocumentProxy = _pdfjs.PDFDocumentProxy;
+  export import DocumentInitParameters = _pdfjs.DocumentInitParameters;
+  export import PDFPageProxy = _pdfjs.PDFPageProxy;
+  export import PDFDocumentLoadingTask = _pdfjs.PDFDocumentLoadingTask;
+  export import PageIndex = _pdfjs.PageIndex;
+  export import Outline = _pdfjs.Outline;
+  export import Transform = _pdfjs.Transform;
+  export import Point = _pdfjs.Point;
+  export import Rect = _pdfjs.Rect;
+  export import PDFObjects = _pdfjs.PDFObjects;
+  export import PDFOperatorList = _pdfjs.PDFOperatorList;
+  export import PageViewport = _pdfjs.PageViewport;
+  export import Annotation = _pdfjs.Annotation;
+  export import RenderParameters = _pdfjs.RenderParameters;
+  export import PDFViewerOptions = _pdfjs.PDFViewerOptions;
+  export import DocumentLoadingProgress = _pdfjs.DocumentLoadingProgress;
+
+  //
+  // All other types. Warning: this should only be used to access types, not
+  // functions or variables.
+  //
+  export import types = _pdfjs;
+}
+
+declare module 'pdfjs-dist' {
```

Comment:
I would call this `pdfjs-dist/web/pdf_viewer`.
This conforms with the `webpackUniversalModuleDefinition` in the built `pdf_viewer.js`.
I can actually use it with a `pdfjs-dist/web/pdf_viewer/webpack` module and Webpack to automatically import the file.

Reply:
@Snuffleupagus Could you comment on what module the viewer components belong in, if any? It looks like @nicolas-schreiber is right that in `dist/web/pdf_viewer.js` there is a module definition for `pdfjs-dist/web/pdf_viewer` but the official download (https://mozilla.github.io/pdf.js/getting_started/#download) doesn't have this file at all, instead it has `web/viewer.js` and no module definition for `pdf_viewer`.

I'm unsure what the right thing is here.